### DISCUSSION TITLE
METviewer Performance Diagram issue

### DISCUSSION THREAD

**Discussion by ChrisMacIntosh-NOAA** [dtcenter/METplus#1436] [Plot Generation] [answered]
Trying to get METviewer GUI to plot performance diagrams and I'm having no luck. POD and FAR plot can be plotted on their own without issue.

I followed the template for precip (this is for snow, but I can't get it to work for APCP either). XML is below.

Any thoughts?

<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<plot_spec>
    <connection>
        <host>[metviewer-dev-2-cluster.cluster-c0bl5kb6fffo.us-east-1.rds.amazonaws.com](http://metviewer-dev-2-cluster.cluster-c0bl5kb6fffo.us-east-1.rds.amazonaws.com/)</host>
        <database>mv_snow_test</database>
        <user>******</user>
        <password>******</password>
        <management_system>aurora</management_system>
    </connection>
    <folders>
        <r_tmpl>/var/lib/tomcat/webapps/metviewer//R_tmpl</r_tmpl>
        <r_work>/var/lib/tomcat/webapps/metviewer//R_work</r_work>
        <plots>/var/www/output//plots</plots>
        <data>/var/www/output//data</data>
        <scripts>/var/www/output//scripts</scripts>
    </folders>
    <plot>
        <template>performance.R_tmpl</template>
        <series1>
            <field name="model">
                <val>CONUSARW</val>
                <val>CONUSARW2</val>
                <val>CONUSFV3</val>
            </field>
        </series1>
        <plot_fix>
            <field equalize="false" name="fcst_var">
                <set name="fcst_var_0">
                    <val>WEASD_24</val>
                </set>
            </field>
            <field equalize="false" name="fcst_lead">
                <set name="fcst_lead_1">
                    <val>240000</val>
                    <val>360000</val>
                    <val>480000</val>
                </set>
            </field>
            <field equalize="false" name="vx_mask">
                <set name="vx_mask_2">
                    <val>MDW</val>
                </set>
            </field>
            <field equalize="false" name="fcst_valid_beg">
                <set name="fcst_valid_beg_3">
                    <val>2022-02-03 00:00:00</val>
                </set>
            </field>
        </plot_fix>
        <plot_cond/>
        <annotation_template/>
        <indep name="fcst_thresh">
            <val label="&gt;=2.54" plot_val="">&gt;=2.54</val>
        </indep>
        <agg_stat>
            <agg_ctc>true</agg_ctc>
            <boot_repl>1</boot_repl>
            <boot_random_seed/>
            <boot_ci>perc</boot_ci>
            <eveq_dis>false</eveq_dis>
            <cache_agg_stat>false</cache_agg_stat>
        </agg_stat>
        <plot_stat>median</plot_stat>
        <tmpl>
            <data_file>plot_20220210_151137.data</data_file>
            <plot_file>plot_20220210_151137.png</plot_file>
            <r_file>plot_20220210_151137.R</r_file>
            <title>test title</title>
            <x_label>test x_label</x_label>
            <y1_label>test y_label</y1_label>
            <y2_label/>
            <caption/>
            <job_title/>
            <keep_revisions>false</keep_revisions>
            <listdiffseries1>list()</listdiffseries1>
            <listdiffseries2>list()</listdiffseries2>
        </tmpl>
        <execution_type>Rscript</execution_type>
        <event_equal>false</event_equal>
        <vert_plot>false</vert_plot>
        <x_reverse>false</x_reverse>
        <num_stats>false</num_stats>
        <indy1_stag>false</indy1_stag>
        <indy2_stag>false</indy2_stag>
        <grid_on>true</grid_on>
        <sync_axes>false</sync_axes>
        <dump_points1>false</dump_points1>
        <dump_points2>false</dump_points2>
        <log_y1>false</log_y1>
        <log_y2>false</log_y2>
        <varianceinflationfactor>true</varianceinflationfactor>
        <plot_type>png16m</plot_type>
        <plot_height>8.5</plot_height>
        <plot_width>11</plot_width>
        <plot_res>72</plot_res>
        <plot_units>in</plot_units>
        <mar>c(8,4,5,4)</mar>
        <mgp>c(1,1,0)</mgp>
        <cex>1</cex>
        <title_weight>2</title_weight>
        <title_size>1.4</title_size>
        <title_offset>-2</title_offset>
        <title_align>0.5</title_align>
        <xtlab_orient>1</xtlab_orient>
        <xtlab_perp>-0.75</xtlab_perp>
        <xtlab_horiz>0.5</xtlab_horiz>
        <xtlab_freq>0</xtlab_freq>
        <xtlab_size>1</xtlab_size>
        <xlab_weight>1</xlab_weight>
        <xlab_size>1</xlab_size>
        <xlab_offset>2</xlab_offset>
        <xlab_align>0.5</xlab_align>
        <ytlab_orient>1</ytlab_orient>
        <ytlab_perp>0.5</ytlab_perp>
        <ytlab_horiz>0.5</ytlab_horiz>
        <ytlab_size>1</ytlab_size>
        <ylab_weight>1</ylab_weight>
        <ylab_size>1</ylab_size>
        <ylab_offset>-2</ylab_offset>
        <ylab_align>0.5</ylab_align>
        <grid_lty>3</grid_lty>
        <grid_col>#cccccc</grid_col>
        <grid_lwd>1</grid_lwd>
        <grid_x>listX</grid_x>
        <x2tlab_orient>1</x2tlab_orient>
        <x2tlab_perp>1</x2tlab_perp>
        <x2tlab_horiz>0.5</x2tlab_horiz>
        <x2tlab_size>0.8</x2tlab_size>
        <x2lab_size>0.8</x2lab_size>
        <x2lab_offset>-0.5</x2lab_offset>
        <x2lab_align>0.5</x2lab_align>
        <y2tlab_orient>1</y2tlab_orient>
        <y2tlab_perp>0.5</y2tlab_perp>
        <y2tlab_horiz>0.5</y2tlab_horiz>
        <y2tlab_size>1</y2tlab_size>
        <y2lab_size>1</y2lab_size>
        <y2lab_offset>1</y2lab_offset>
        <y2lab_align>0.5</y2lab_align>
        <legend_box>o</legend_box>
        <legend_inset>c(0, -.25)</legend_inset>
        <legend_ncol>3</legend_ncol>
        <legend_size>0.8</legend_size>
        <caption_weight>1</caption_weight>
        <caption_col>#333333</caption_col>
        <caption_size>0.8</caption_size>
        <caption_offset>3</caption_offset>
        <caption_align>0</caption_align>
        <ci_alpha>0.05</ci_alpha>
        <eqbound_low>-0.001</eqbound_low>
        <eqbound_high>0.001</eqbound_high>
        <plot_ci>c("none","none","none")</plot_ci>
        <show_signif>c(FALSE,FALSE,FALSE)</show_signif>
        <plot_disp>c(TRUE,TRUE,TRUE)</plot_disp>
        <colors>c("#ff0000FF","#00ff7fFF","#8000ffFF")</colors>
        <pch>c(20,20,20)</pch>
        <type>c("b","b","b")</type>
        <lty>c(1,1,1)</lty>
        <lwd>c(1,1,1)</lwd>
        <con_series>c(1,1,1)</con_series>
        <order_series>c(1,2,3)</order_series>
        <plot_cmd/>
        <legend>c("","","")</legend>
        <create_html>TRUE</create_html>
        <y1_lim>c()</y1_lim>
        <x1_lim>c()</x1_lim>
        <y1_bufr>0.04</y1_bufr>
        <y2_lim>c()</y2_lim>
    </plot>
</plot_spec>

**Comment by TaraJensen** [comment]
Chris,

I see you need assistance with METviewer.  Here is a link to the online documentation [https://metviewer.readthedocs.io/en/latest/Users_Guide/index.html](https://metviewer.readthedocs.io/en/latest/Users_Guide/index.html), which includes a chapter on the Performance Diagram.  Please take a look at that and see if that answers your question. 

Hope this helps.

**Comment by ChrisMacIntosh-NOAA** [comment]
Hey Tara,

I've looked through the documentation and that has not solved my issue here.

Thanks!

**Reply by TaraJensen** [reply]
@TatianaBurek - can you take a look at this?

**Comment by TatianaBurek** [comment, marked as answer]
Chris,
When you create a performance diagrams you select Agg stat -> CTC.
In this case METviewer retrieves data  for FAR and PODY from line_data_cts.  The table 'line_data_cts' in mv_snow_test database contains data only for  APCP_01 , APCP_06, APCP_03, APCP_24  and not for WEASD_24. That is why the diagram is empty.

**Reply by jprestop** [reply]
Hi @ChrisMacIntosh-NOAA.  I just wanted to follow up and ensure that @TatianaBurek answered your question? If not, please let us know what other questions you have.  If so, could you please select her response as the answer?  Then, we'll go ahead and lock this Discussion. We want to encourage users to ask new questions in new discussions rather than posting to old ones. Hopefully that'll make the questions/answers easier for other users to follow. So if/when more issues/questions arise, please feel free to start a new discussion.  Thanks!

**Reply by jprestop** [reply]
HI @ChrisMacIntosh-NOAA.  I just wanted to follow up again and ensure that your question was answered?

**Reply by ChrisMacIntosh-NOAA** [reply]
It did! I've selected that as the answer.
